.github/workflows/build_go_docker_image.yml

```diff
@@ -25,17 +25,19 @@ jobs:
       GITHUB_TOKEN: ${{ secrets.CIPIPELINE_NEXUS_PASSWORD }}
 
     steps:
-      - name: Get branch name
-        shell: bash
-        run: |
-          echo "SOURCE_BRANCH=$(echo ${GITHUB_REF#refs/heads/} | tr / -)" >> $GITHUB_ENV
-          echo "DOCKER_IMAGE_NAME=$(echo ${GITHUB_REPOSITORY} | cut -f2 -d'/')" >> $GITHUB_ENV
 
       - name: Checkout code
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
+      - name: Get branch name
+        shell: bash
+        run: |
+          echo "SOURCE_BRANCH=$(echo ${GITHUB_REF#refs/heads/} | tr / -)" >> $GITHUB_ENV
+          DOCKER_IMAGE_NAME=$(yq '.service-name' setup.yaml)
```

Comment:
Why are we depending on setup.yaml? In case image name is different than repo name, we should be reading from convoy.yaml. 

Secondly, what is the impact of this change on other repositories ? Does all repositories have the image name defined in `setup.yaml`?

Reply:
We can use either of convoy.yaml or setup.yaml but since build using setup reads value from setup.yaml it better suited to here also read it from there itself.

There won't be any impact on other repos since their build is also using value from setup.yaml which is defined.